tests/server/test_env_settings.py

```diff
@@ -225,3 +225,56 @@ async def assert_clear_env(env_id: str, clear_succeeds: bool) -> None:
     result = await client.environment_settings_set(env_id, data.PROTECTED_ENVIRONMENT, False)
     assert result.code == 200
     await assert_clear_env(env_id, clear_succeeds=True)
+
+
+@pytest.mark.asyncio
+async def test_decommission_protected_environment(server, client):
+    result = await client.create_project("env-test")
+    assert result.code == 200
+    project_id = result.result["project"]["id"]
+
+    result = await client.create_environment(project_id=project_id, name="dev")
+    assert result.code == 200
+    env_id = result.result["environment"]["id"]
+
+    async def push_version_to_environment() -> None:
+        res = await client.reserve_version(env_id)
+        assert res.code == 200
+        version = res.result["data"]
+
+        result = await client.put_version(
+            tid=env_id,
+            version=version,
+            resources=[],
+            unknowns=[],
+            compiler_version=get_compiler_version(),
+        )
+        assert result.code == 200
+
+    async def assert_decomission_env(env_id: str, decommission_succeeds: bool) -> None:
+        result = await client.list_versions(env_id)
+        assert result.code == 200
+        assert len(result.result["versions"]) != 0
+        # Execute clear operation
+        result = await client.environment_decommission(env_id)
+        assert result.code == 200 if decommission_succeeds else 403
+        # Assert result
+        result = await client.list_versions(env_id)
+        assert result.code == 200
+        # Original and the empty for decommissioning
+        assert ("version_info" in result.result["versions"][0]) == decommission_succeeds
```

Comment:
This assertion needs a bit more info. From the comment I would expect that the assertion compares the number of versions before the decommissioning with the number of version after the decommissioning, but it doesn't.